File: python/src/Structural/Proxy/PotOfTeaProxy.py

# the Proxy

from PotOfTeaInterface import PotOfTeaInterface 
from PotOfTea import PotOfTea

class PotOfTeaProxy(PotOfTeaInterface):

	potOfTea = None

	def pourTea(self) :
		self.potOfTea = PotOfTea()
		return self.potOfTea.pourTea()

File: ruby/src/Structural/Proxy/PotOfTeaProxy.rb

# the Proxy

#from PotOfTeaInterface import PotOfTeaInterface 
#from PotOfTea import PotOfTea

require './src/Structural/Proxy/PotOfTeaInterface.rb'
require './src/Structural/Proxy/PotOfTea.rb'

class PotOfTeaProxy < PotOfTeaInterface

	def initialize
		@potOfTea = PotOfTea.new
	end

	def pourTea
		unless @potOfTea
			@potOfTea = PotOfTea.new
		end
		@potOfTea.pourTea
	end

end

